cli/src/auth/google_cloud.rs

```diff
@@ -74,6 +57,63 @@ const AUTH_URI: &str = "https://accounts.google.com/o/oauth2/auth";
 const REDIRECT_URI: &str = "http://127.0.0.1/8855";
 const AUTH_PROVIDER_X509_CERT_URL: &str = "https://www.googleapis.com/oauth2/v1/certs";
 
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct GoogleCloudConfig {
+    /// used when authorizing calls to oauth2 enabled services.
+    pub access_token: String,
+    /// used to refresh an expired access_token.
+    pub refresh_token: String,
+    /// The time when the token expires.
+    pub expiry_time: String,
+    /// Optionally included by the OAuth2 server and may contain information to verify the identity
+    /// used to obtain the access token.
+    /// Specifically Google API:s include this if the additional scopes "email" and/or "profile"
+    /// are used. In that case the content is an JWT token.
+    pub id_token: Option<String>,
+}
+
+struct ConfigTokenStore {
+    config: Config,
+}
+
+#[async_trait]
+impl TokenStorage for ConfigTokenStore {
+    async fn set(&self, _scopes: &[&str], token: TokenInfo) -> anyhow::Result<()> {
+        let mut config = self.config.clone();
+
+        config.auth.google_cloud = Some(GoogleCloudConfig {
+            access_token: token.access_token.expect("Invalid access token"),
+            refresh_token: token.refresh_token.expect("Invalid refresh token"),
+            expiry_time: token
+                .expires_at
+                .expect("Invalid expiry time")
+                .format(&format_description::well_known::Iso8601::DEFAULT)?,
```

Comment:
any reason why you want to use this format? okta and vault are using RFC3339
- https://github.com/mindvalley/wukong-cli/blob/main/cli/src/auth/okta.rs#L235
- https://github.com/mindvalley/wukong-cli/blob/main/cli/src/auth/vault.rs#L231

Reply:
Oh wait, i thought both okta and vault are using ISO8601, but i'll change this to RFC339